### PR TITLE
refactor: Split Checker as an individual jar

### LINKED ISSUE #247: Package the checker classes into a jar

### PR DESCRIPTION
**Pull request by Vigilans**
Fix #247

### CODE REVIEW

File: jdtls.ext/com.shengchen.checkstyle.checker/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+				<configuration>
+					<configLocation>${project.parent.basedir}/config/checkstyle.xml</configLocation>
+					<propertyExpansion>checkstyle.header.file=${project.parent.basedir}/config/java.header</propertyExpansion>
+					<failOnViolation>true</failOnViolation>
+				</configuration>
+			</plugin>
+		</plugins>
+	</build>
+	<properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (jdneo) [inline jdtls.ext/com.shengchen.checkstyle.checker/pom.xml]: Uniform the indention

File: jdtls.ext/com.shengchen.checkstyle.checker/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	<name>${base.name} :: Checker</name>
+	<dependencies>
+		<dependency>
+			<groupId>com.puppycrawl.tools</groupId>
+			<artifactId>checkstyle</artifactId>
+			<version>8.18</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.shengchen.checkstyle</groupId>
```
Comment (jdneo) [inline jdtls.ext/com.shengchen.checkstyle.checker/pom.xml]: Why we need this dependency?

Comment (Vigilans) [reply jdtls.ext/com.shengchen.checkstyle.checker/pom.xml]: `checker` need to reference classes in `runner.api`, such as `CheckResult` and `ICheckerService`.

Comment (jdneo) [reply jdtls.ext/com.shengchen.checkstyle.checker/pom.xml]: ok

**Pull request merged**